ietf/draft-ietf-jmap-calendars-18.txt

```diff
@@ -0,0 +1,3360 @@
+
+      Defines the sort order of calendars when presented in the client's
+      UI, so it is consistent between devices.  The number MUST be an
+      integer in the range 0 <= sortOrder < 2^(31.)
+
+      A calendar with a lower order is to be displayed before a calendar
+      with a higher order in any list of calendars in the client's UI.
+      Calendars with equal order should be sorted in alphabetical order
+      by name.  The sorting should take into account locale-specific
+      character order convention.
+
+   *  *isSubscribed*: Boolean
+
+      True if the user has indicated they wish to see this Calendar in
+      their client.  This SHOULD default to false for Calendars in
+      shared accounts the user has access to and true for any new
+      Calendars created by the user themself.
+
+      If false, the calendar SHOULD only be displayed when the user
+      explicitly requests it or to offer it for the user to subscribe
+      to.  For example, a company may have a large number of shared
+      calendars which all employees have permission to access, but the
+      user would only subscribe to the ones they care about and want to
+      be able to have normally accessible.
+
+
+
+Jenkins & Douglass       Expires 9 December 2024               [Page 13]
+
+Internet-Draft               JMAP Calendars                    June 2024
+
+
+   *  *isVisible*: Boolean (default: true)
+
+      If true, the calendar's events should be displayed to the user.
+      If false, the user has indicated they wish to temporarily hide
+      this calendar's events.  Clients MUST ignore this property if
+      isSubscribed is false.  If an event is in multiple calendars, it
+      should be displayed if isVisible is true for any of those
+      calendars.
+
+   *  *isDefault*: Boolean (server-set)
+
+      This SHOULD be true for exactly one calendar in any account, and
+      MUST NOT be true for more than one calendar within an account.
+      The default calendar should be used by clients whenever they need
+      to choose a calendar for the user within this account, and they do
+      not have any other information on which to make a choice.  For
+      example, if the user creates a new event, the client may
+      automatically set the event as belonging to the default calendar
+      from the user's primary account.
+
+   *  *includeInAvailability*: String
+
+      This determines which events in this calendar will be used as part
+      of availability calculation.  The value MUST be one of:
+
+      -  all: all events are considered.
+      -  attending: events the user is a confirmed or tentative
+         participant of are considered.
+      -  none: all events are ignored (but may be considered if also in
+         another calendar).
+
+      This should default to "all" for the calendars in the user's own
+      account, and "none" for calendars shared with the user.
+
+   *  *defaultAlertsWithTime*: Id[Alert]|null
```

Comment:
It might be nice to discuss that defaultAlertsWithTime/showWithoutTime false would be for events that happen at specific times, and defaultAlertsWithoutTime/showWithoutTime true would be for all day events, etc...

Without looking at the JSCalendar spec I found understanding these options a little confusing

Reply:
I've added *(commonly referred to as "all day" events)* to the description of `defaultAlertsWithoutTime`, which hopefully helps a little with this.